### DISCUSSION TITLE
Cannot find and install atoti-plus

### DISCUSSION THREAD

**Discussion by NZepeda** [atoti/atoti#574] [Q&A] [answered]
Hello, I'm attempting to create a docker container that utilizes `atoti-plus`. In the past, I was able to install `atoti-plus`, build the container, and deploy it just fine. Recently, I've attempted to deploy an updated version of the container, however, now the atoti plus installation fails with the following error: 

```
#5 1.583 Looking in indexes: https://pypi.org/simple, https://nestor.zepeda:****@pypi.atoti.io/simple
#5 2.263 ERROR: Could not find a version that satisfies the requirement atoti-plus==0.6.4 (from versions: none)
#5 2.263 ERROR: No matching distribution found for atoti-plus==0.6.4
```

The command I am running is:

```
pip install atoti-plus==0.6.4 --extra-index-url https://${ATOTI_PLUS_USERNAME}:${ATOTI_PLUS_PASSWORD}@pypi.atoti.io/simple --no-cache-dir
```

I am getting the exact same error even when I don't specify a version of the `atoti-plus` package for installation. 

I see there is a similar issue brought up in this [discussion](https://github.com/atoti/atoti/discussions/566), however, the same fix does not seem to alleviate my current issue. 

The version of Python I am running is `3.10.4` and pip version is `22.0.4`.

**Comment by tibdex** [comment]
Hi Nestor,

> Hello, I'm attempting to create a docker container that utilizes atoti-plus

Sounds like you would like to use our Project template, and in particular, its [ `atoti-plus` branch](https://github.com/atoti/project-template/tree/atoti-plus). This [command](https://github.com/atoti/project-template/blob/ac841fa1725b364b10e7e9e077b7a4deb144a61b/cli/app.py#L20-L39) shows how to build the Docker image.

> The version of Python I am running is 3.10.4

The project template uses Python 3.9 so it would be easier to use this version too.

**Reply by NZepeda** [reply]
Hey @tibdex, 

Thanks for the response. I'll take a look at your project template. However, I'm still a bit confused as to why I can't install `atoti-plus` directly. Did the location of the package change possibly? It seems odd that the installation command that used to work as expected, is no longer working for **_any_** version of `atoti-plus`. I get the above error regardless of whether I specify a version or not. According to the error, it seems like there are no versions of `atoti-plus` available to install. I've also even tried downgrading my version of python to `v3.9.12` and I'm still seeing the error.

**Reply by tibdex** [reply]
By any chance, are you on macOS with an Apple silicon chip? Because I don't think `jdk4py` supports this architecture.

**Reply by NZepeda** [reply]
@tibdex Nope. I am on MacOS but it's an Intel chip model.

**Comment by NZepeda** [comment, marked as answer]
I found the issue, which turned out to be an extreme edge case that I suspect will be very rare that others run into but it's worthing noting in case it comes up again. **The issue was that somehow, my account on Artifactory got locked preventing me from being able to download any artifacts from Artifactory**. I didn't realize it was locked because using the exact same credentials on Confluence/Jira worked just fine. It wasn't until I tried accessing an AUI-4 artifact, that I realized my credentials weren't working for Artifactory. My confusion around this was compounded even more since there wasn't any error messaging regarding this when installing. Perhaps that's something that can be improved? 

After consulting with Loïc to get my account unlocked, the `atoti-plus` installation worked fine.

**Reply by tibdex** [reply]
Thanks for posting this detailed answer 👍.
